.github/workflows/ci.yaml

```diff
@@ -444,3 +445,65 @@ jobs:
         asset_name: tanzu-apps-plugin${{ matrix.os-arch }}-${{ env.RELEASE_VERSION }}.tar.gz
         asset_content_type: application/gzip
 
+  publish-oci-artifact:
+    env:
+      TKG_CUSTOM_IMAGE_REPOSITORY: projects-stg.registry.vmware.com/tkg
+      TANZU_CLI_NO_INIT: true
+      TANZU_HOME: $HOME/tanzu
+      ARTIFACTS_DIR: ./artifacts
+    needs:
+    - release
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: vmware-tanzu/carvel-setup-action@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+    - name: Install tanzu cli
+      run: |
+        TANZU_VERSION=$(cat TANZU_VERSION)
+        env
+        mkdir -p ${TANZU_HOME}
+        curl -Lo tanzu-framework.tar.gz https://github.com/vmware-tanzu/tanzu-framework/releases/download/${TANZU_VERSION}/tanzu-framework-linux-amd64.tar.gz
+        tar -xzf tanzu-framework.tar.gz -C ${TANZU_HOME}
+        sudo mv ${TANZU_HOME}/cli/core/${TANZU_VERSION}/tanzu-core-linux_amd64 /usr/local/bin/tanzu
+        chmod +x /usr/local/bin/tanzu
+        tanzu init
+        tanzu plugin repo add -b tanzu-cli-admin-plugins -n admin -p artifacts-admin
+        curl -Lo admin-plugins.tar.gz https://github.com/vmware-tanzu/tanzu-framework/releases/download/${TANZU_VERSION}/tanzu-framework-plugins-admin-linux-amd64.tar.gz
+        tar -xzf admin-plugins.tar.gz -C ${TANZU_HOME}
+        tanzu plugin install builder --local ${TANZU_HOME}/admin-plugins
+        tanzu plugin list
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Set repos
+      run: |
+        echo "DISTRIBUTION_REPO=ghcr.io/${{ github.repository }}/distribution" >> $GITHUB_ENV
+        echo "DISCOVERY_REPO=ghcr.io/${{ github.repository }}/discovery" >> $GITHUB_ENV
+        echo $DISTRIBUTION_REPO
+        echo $DISCOVERY_REPO
+    - name: Build Version
+      run: |
+        echo "BUILD_VERSION=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+    - name: Download build artifacts
+      uses: actions/download-artifact@v3
+      with:
+        name: tanzu-apps-plugin-build.tar.gz
+    - name: Get build artifact
+      run: |
+        set -o errexit
+        set -o nounset
+        set -o pipefail
+
+        if [ ! -d ${ARTIFACTS_DIR} ]; then
+          mkdir -p ${ARTIFACTS_DIR}
+        fi
+
+        tar -xvf tanzu-apps-plugin-build.tar.gz -C ${ARTIFACTS_DIR}
```

Comment:
by doing this , will there be any issue to the data already present in the artifact directory?

Reply:
by that point, the `artifacts` directory does not exist, I'm creating it and unzipping the files there. Check the `.gitignore` file